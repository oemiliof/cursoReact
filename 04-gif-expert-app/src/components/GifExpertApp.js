import React, { useState } from 'react'
import AddCategory from './Categorias/AddCategory';
import GifGrid from './Categorias/GifGrid';

export const GifExpertApp = () => {

    // const categories = ['One Punch', 'Samurai X', 'Fragon Ball'];

    const [categories, setCategories] = useState(['One Punch'])

    // const handleAdd = () => {
    //     // setCategories([...categories, ('test ' + parseInt(categories.length + 1))]);
    //     // setCategories([('test ' + parseInt(categories.length + 1)), ...categories]);
    //     setCategories(cats => [...categories, ('test ' + parseInt(categories.length + 1))]);



    //     console.log(categories);
    // };

    // console.log(categories)

    return (
        <>
            <h2>GifExpertApp</h2>

            <AddCategory setCategories={setCategories} />

            <hr />

            {/* <button onClick={handleAdd}>Agregar</button> */}

            {
                categories.map(category => (
                    <GifGrid
                        key={category}
                        category={category}
                    />
                ))
            }
        </>
    )
}
