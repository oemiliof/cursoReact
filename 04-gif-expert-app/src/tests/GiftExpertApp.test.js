import react from "react";
import "jest";
import { shallow } from "enzyme";
import { GifExpertApp } from "../components/GifExpertApp";

describe("Pruebas de <GifExpertApp/>", () => {
  test("debe de mostrarse correctamente", () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de mostrar una lista de categorias", () => {
    const categories = ["One Punch", "Dragon Ball"];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifGrid").length).toBe(categories.length);
  });
});
