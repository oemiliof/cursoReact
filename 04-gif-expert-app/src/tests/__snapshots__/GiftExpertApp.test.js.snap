// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <GifExpertApp/> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <GifGrid
    category="One Punch"
    key="One Punch"
  />
  <GifGrid
    category="Dragon Ball"
    key="Dragon Ball"
  />
</Fragment>
`;

exports[`Pruebas de <GifExpertApp/> debe de mostrarse correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
</Fragment>
`;
